ts/test/functional/query-deployments.spec.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * Functional tests for querying deployments using the Akash Chain SDK
+ * 
+ * These tests demonstrate how to query live deployment data from the Akash network.
+ */
+
+import { describe, expect, it } from "@jest/globals";
+import Long from "long";
+
+import { createChainNodeSDK } from "../../src/sdk/chain/server/index.ts";
+import { validateProtobufDeserialization } from "@test/helpers/protobuf-validation";
+// Import actual protobuf types for better type safety
+import type { 
+  QueryDeploymentsResponse,
+  QueryDeploymentResponse 
+} from "../../src/generated/protos/akash/deployment/v1beta4/query.ts";
+
+describe("Deployment Queries", () => {
+  // Use the working configuration from your provided snippet
+  // Query and TX endpoints are different!
+  // Note: These are gRPC endpoints that need proper URL schemes
+  const QUERY_RPC_URL = process.env.QUERY_RPC_URL || "http://rpc.dev.akash.pub:30090";
+  const TX_RPC_URL = process.env.TX_RPC_URL || "https://testnetrpc.akashnet.net:443";
+  const TEST_TIMEOUT = 15000;
+
+  // Type-safe validator for deployment responses using the actual protobuf type
+  const validateDeploymentResponseDeserialization = (deploymentResponse: any) => {
+    // This provides full type safety with the actual protobuf interface
+    validateProtobufDeserialization<QueryDeploymentResponse>(deploymentResponse);
+    
+    expect(deploymentResponse.deployment).toBeDefined();
+    expect(Array.isArray(deploymentResponse.groups)).toBe(true);
+    expect(deploymentResponse.escrowAccount).toBeDefined();
+  };
+
+  // Helper function to create SDK instance
+  const createTestSDK = () => createChainNodeSDK({
+    query: { baseUrl: QUERY_RPC_URL },
+    tx: { baseUrl: TX_RPC_URL, signer: null as any },
+  });
+
+  // Helper function to create pagination config
+  const createPagination = (limit: number) => ({
+    key: new Uint8Array(0),
+    offset: Long.UZERO,
+    limit: Long.fromNumber(limit),
+    countTotal: false,
+    reverse: false,
+  });
+
+  // Helper function to create empty filters
+  const createEmptyFilters = () => ({
+    owner: "",
+    dseq: Long.UZERO,
+    state: "",
+  });
+
+  // Helper function to handle response (deployments is unary, returns Promise not AsyncIterable)
+  const getResponse = async (responsePromise: Promise<any>) => {
+    const response = await responsePromise;
+    return response;
+  };
+
+  it("should query deployments from the network", async () => {
+    const sdk = createTestSDK();
+
+    const queryParams = {
+      filters: createEmptyFilters(),
+      pagination: createPagination(10),
+    };
+
+    const responsePromise = sdk.akash.deployment.v1beta4.getDeployments(queryParams);
+
+    const response = await getResponse(responsePromise);
```

Comment:
```suggestion
    const response = await sdk.akash.deployment.v1beta4.getDeployments(queryParams);
```

Reply:
I'm using the helper getResponse to return a promise not an AsyncIterable.